ui/desktop/app/routes/scopes/scope/projects/targets/target.js

```diff
@@ -5,21 +5,105 @@
 
 import Route from '@ember/routing/route';
 import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { loading } from 'ember-loading';
 
 export default class ScopesScopeProjectsTargetsTargetRoute extends Route {
   // =services
 
   @service store;
+  @service confirm;
+  @service ipc;
+  @service router;
+  @service session;
+
+  queryParams = {
+    isConnecting: {
+      refreshModel: true,
+    },
+  };
 
   // =methods
 
   /**
    * Load a target
    * @param {object} params
    * @param {string} params.target_id
+   * @param {bool} params.isConnecting
    * @return {TargetModel}
    */
-  model({ target_id }) {
-    return this.store.findRecord('target', target_id, { reload: true });
+  async model({ target_id, isConnecting }) {
+    const target = await this.store.findRecord('target', target_id, {
+      reload: true,
+    });
+
+    if (isConnecting) {
+      this.preConnect(target);
+    }
+    return target;
+  }
+
+  /**
+   * Determine if connection workflow based on target attributes.
```

Comment:
None blocking comment: What "if connection workflow" means? I assume IF has target.address does connect directly to the target, so connection workflow means showing the choose a host modal?

Reply:
This is a typo that I meant to fix. I will update this shortly.

Reply:
Can you also explain in the comment why this pre-connect workflow is needed? I don't think it's evident in this PR as we aren't dealing with hosts right now

Reply:
Sorry, I hadn't refreshed before resolving this conversation. This pre-connect might be changing as I add in the host logic. So I will update this in the next PR if that works with you?

Reply:
Yeah that's fine, I was just saying in this PR the comment would be helpful since without knowing the followup PRs the pre-connect logic seems unnecessary